### PR TITLE
Changes in action and reducer

### PR DESCRIPTION
**Pull request by bluwhalee**
Added redux, made common components, (you can update Todo list through input field)

### CODE REVIEW

File: src/components/Form.jsx [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+import { v4 as uuid } from 'uuid';
 
 export function Form() {
     const [task, setTask] = useState('');
-
+    const dispatch = useDispatch();
     function handleSubmit(event) {
         event.preventDefault();
+        const unique_id = uuid();
+        const small_id = unique_id.slice(0,8)
```
Comment (IshfaqAhmedAwan) [inline src/components/Form.jsx]: why are we slicing the unique Id? if there is no. specific reason then dont slice it

File: src/components/Todo.jsx [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import {useState} from "react";
 import {useSelector} from "react-redux";
 import {operationsReducer} from "../redux/todo/reducers/operations.js";
+import Icon, {icon} from "react-icons-kit"
+import {trash} from "react-icons-kit/feather/trash"
```
Comment (IshfaqAhmedAwan) [inline src/components/Todo.jsx]: cant we use just fontawesome icons?

File: src/components/Todo.jsx [hunk 3]
```diff
@@ -9,10 +12,13 @@ export function Todo() {
 
     return (todos.map((todo)=>(
           <div key={todo.id}>
+              <input type="checkbox" checked={todo.status}></input>
```
Comment (IshfaqAhmedAwan) [inline src/components/Todo.jsx]: make a separate common component for input field, considering there can be more than one input fields.

File: src/redux/todo/reducers/operations.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-
+import {ADD_TODO} from "../actions/index.js";
 
 const dummy = [
     {id:23, task: "Jomoloko", status:false},
```
Comment (IshfaqAhmedAwan) [inline src/redux/todo/reducers/operations.js]: give this constant a more useful name

**Review by IshfaqAhmedAwan — CHANGES_REQUESTED** [review]

File: src/components/Todo.jsx (hunk 2 above)
Comment (bluwhalee) [reply src/components/Todo.jsx]: tried them first, werent working somehow, so moved to this

File: src/components/Todo.jsx (hunk 3 above)
Comment (bluwhalee) [reply src/components/Todo.jsx]: can you please eleborate on this a bit.

File: src/components/Todo.jsx (hunk 2 above)
Comment (IshfaqAhmedAwan) [reply src/components/Todo.jsx]: try to make them work

File: src/components/Todo.jsx (hunk 3 above)
Comment (IshfaqAhmedAwan) [reply src/components/Todo.jsx]: as we discussed in meeting

File: src/components/common/inputComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const InputComponent = (props) =>
+{
+    return(
+        <input type={props.type} value={props.value} onChange={props.HandleOnChange} placeholder={props.placeholder} />
+    )
```
Comment (IshfaqAhmedAwan) [inline src/components/common/inputComponent.jsx]: lets use material ui for input

File: src/components/AddTodo.jsx [hunk 6]
```diff
@@ -23,8 +30,8 @@ export function Form() {
             <h1>Todo App</h1>
             <form onSubmit={handleSubmit}>
                 <label>
-
-                    <input type="text" value={task} onChange={handleChange} placeholder="Add new task" />
+                    <InputComponent  type="text" value={task} HandleOnChange={handleChange} placeholder="Add new task" />
+                    {/*<input type="text" value={task} onChange={handleChange} placeholder="Add new task" />*/}
```
Comment (IshfaqAhmedAwan) [inline src/components/AddTodo.jsx]: remove the useles comment

**Review by IshfaqAhmedAwan — APPROVED** [review]

File: src/components/Todo.jsx [hunk 7]
```diff
@@ -1,22 +1,35 @@
...
+                  <FontAwesomeIcon icon={faPen}></FontAwesomeIcon>
 
               </div>
           </div>
         )
-    ))
+    )}
+        </div>)
 
 }
```
Comment (ahmadfaraz629) [inline src/components/Todo.jsx]: @bluwhalee  you can see a red error icon means that you have an extra space the the end of the file kindly remove it.

File: src/components/common/inputComponent.jsx [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+}
+export default InputComponent;
```
Comment (ahmadfaraz629) [inline src/components/common/inputComponent.jsx]: Same eof error here.

File: src/redux/todo/actions/index.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+export const ADD_TODO = 'ADD_TODO';
+export const addtodo = (payload) => {
+    return {
+        type:'ADD_TODO',
+        payload
+    }
+}
```
Comment (ahmadfaraz629) [inline src/redux/todo/actions/index.js]: Same eof error here.

**Review by ahmadfaraz629 — COMMENTED** [review]
I also notices that your formating is off in this project. Try to use prettier extension and read about it.

**Comment by ahmadfaraz629** [comment]
@bluwhalee PR description is missing.

Write details of what you did inside this PR. You can add screenshots of your app.

image

**Comment by ahmadfaraz629** [comment]
@bluwhalee fix this typo error
image

**Pull request merged**